packages/z3/z3.4.12.2/opam

```diff
@@ -0,0 +1,43 @@
+opam-version: "2.0"
+maintainer: "weng@cs.jhu.edu"
+authors: "MSR"
+homepage: "https://github.com/Z3prover/z3"
+bug-reports: "https://github.com/Z3prover/z3/issues"
+license: "MIT"
+dev-repo: "git+https://github.com/Z3prover/z3.git"
+depexts: [
+  ["python3-distutils"] {os-family = "debian"}
+]
+build: [
+  [ "python3" "scripts/mk_make.py" "--ml" ]
+  [ make "-C" "build" "-j" jobs ]
+]
+
+install: [
+  [ "sh" "-c" "ocamlfind install z3 build/api/ml/* -dll build/libz3.*"]
+  [ "cp" "build/z3" "%{bin}%/z3"]
+]
+
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "ocamlfind" {build}
+  "zarith"
+  "conf-gmp"
+  "conf-python-3" {build}
+  "conf-c++" {build}
+]
+x-ci-accept-failures: [
+  "centos-7" "oraclelinux-7" # C compiler is too old
```

Comment:
```suggestion
  "archlinux" # C compiler is too old
```
The other distributions are no-longer tested and the build fails on archlinux due to
```
error: ‘uintptr_t’ does not name a type
```
which is an optional type in C++11 or later and makes me believe that the arch image is shipping an old compiler or one with a peculiar configuration

Reply:
https://github.com/Z3Prover/z3/issues/6722
Building with GCC 13 causes the problem, I've encountered that with Fedora 38. The patch is https://github.com/Z3Prover/z3/pull/6723